### PR TITLE
Add commandlets for installing, removing, getting and testing the Azu…

### PR DESCRIPTION
**Pull request by MSSedusch**
…re Enhanced Monitoring extension

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@MSSedusch__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (sedusch). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci test this please

**Comment by hovsepm** [comment]
@MSSedusch please pull from upstream dev branch.

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by MSSedusch** [comment]
I fixed the offline test. Can you rerun the test suite?

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/AEMExtensionTests.ps1 [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+    $loc = Get-ComputeVMLocation
+
+    try
+    {
+        # Setup
+        $vm = Create-AdvancedVM -rgname $rgname -loc $loc
+        $vmname = $vm.Name
+
+		# Get with not extension
+		$extension = Get-AzureRmVMAEMExtension -ResourceGroupName $rgname -VMName $vmname
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/AEMExtensionTests.ps1]: @MSSedusch, please align the indentation in this file.

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+        }
+
+
+
+
+        
+
+
+
+
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs]: please remove white-spaces.

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+                props.MinuteMetrics = new MetricsProperties();
+            }
+            props.MinuteMetrics.RetentionDays = retentionDays;
+            props.MinuteMetrics.MetricsLevel = MetricsLevel.ServiceAndApi;
+
+            if (props.HourMetrics != null && props.HourMetrics.RetentionDays != null)
+            {
+                props.HourMetrics.RetentionDays = retentionDays;
+            }
+            //if (props.Metrics != null && props.Metrics.RetentionDays != null)
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs]: why this code is commented? if it is useless just remove it

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs [hunk 4]
```diff
@@ -0,0 +1,495 @@
...
+
+            Newtonsoft.Json.Linq.JObject jPrivateConfig = new Newtonsoft.Json.Linq.JObject();
+            jPrivateConfig.Add("storageAccountName", new Newtonsoft.Json.Linq.JValue(storageAccountName));
+            jPrivateConfig.Add("storageAccountKey", new Newtonsoft.Json.Linq.JValue(storageAccountKey));
+            jPrivateConfig.Add("storageAccountEndPoint", new Newtonsoft.Json.Linq.JValue(endpoint));
+
+            WriteVerbose("Installing WAD extension");
+            VirtualMachineExtension vmExtParameters = new VirtualMachineExtension();
+
+            //vmExtParameters.Name = AEMExtensionConstants.WADExtensionDefaultName[this.OSType];
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs]: if this is useless just remove it

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs [hunk 5]
```diff
@@ -0,0 +1,495 @@
...
+                
+                var result = Mapper.Map<PSAzureOperationResponse>(op);
+                WriteObject(result);
+            });
+        }
+
+        private VirtualMachine SetAzureVMDiagnosticsExtensionC(VirtualMachine selectedVM, string storageAccountName, string storageAccountKey)
+        {
+            System.Xml.XmlDocument xpublicConfig = null;
+            //var xmlnsConfig = "http://schemas.microsoft.com/ServiceHosting/2010/10/DiagnosticsConfiguration";
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs]: if this is useless just remove it

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs [hunk 6]
```diff
@@ -0,0 +1,495 @@
...
+                    sapmonPublicConfig.Add(new KeyValuePair() { Key = "wad.isenabled", Value = "0" });
+                }
+
+                ExtensionConfig jsonPublicConfig = new ExtensionConfig();
+                jsonPublicConfig.Config = sapmonPublicConfig;
+                //var publicConfString = JsonConvert.SerializeObject(jsonPublicConfig);
+
+                ExtensionConfig jsonPrivateConfig = new ExtensionConfig();
+                jsonPrivateConfig.Config = sapmonPrivateConfig;
+                //var privateConfString = JsonConvert.SerializeObject(jsonPrivateConfig);
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs]: if this is useless just remove it

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs [hunk 7]
```diff
@@ -0,0 +1,495 @@
...
+                    sapmonPublicConfig.Add(new KeyValuePair() { Key = "wad.uri", Value = wadUri });
+                }
+                else
+                {
+                    sapmonPublicConfig.Add(new KeyValuePair() { Key = "wad.isenabled", Value = "0" });
+                }
+
+                ExtensionConfig jsonPublicConfig = new ExtensionConfig();
+                jsonPublicConfig.Config = sapmonPublicConfig;
+                //var publicConfString = JsonConvert.SerializeObject(jsonPublicConfig);
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs]: if this is useless just remove it

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs [hunk 8]
```diff
@@ -0,0 +1,495 @@
...
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            this._Helper = new AEMHelper((err) => this.WriteError(err), (msg) => this.WriteVerbose(msg), (msg) => this.WriteWarning(msg),
+                this.CommandRuntime.Host.UI,
+                AzureSession.ClientFactory.CreateClient<StorageManagementClient>(DefaultProfile.Context, AzureEnvironment.Endpoint.ResourceManager),
+                this.DefaultContext.Subscription);
+            
+
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/SetAzureRmVMAEMExtension.cs]: too many line breaks.

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/RemoveAzureRmVMAEMExtension.cs [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+                        WriteWarning(string.Format(CultureInfo.InvariantCulture, Properties.Resources.AEMExtensionNotFound, this.ResourceGroupName, this.VMName));
+                        return;
+                    }
+                    else
+                    {
+                        this.Name = aemExtension.Name;
+                    }
+                }
+
+
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/RemoveAzureRmVMAEMExtension.cs]: too many line breaks.

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMHelper.cs [hunk 10]
```diff
@@ -0,0 +1,722 @@
...
+                        catch { }
+                    }
+
+                    wait = ((DateTime.Now) - checkStart).TotalMinutes < TimeoutinMinutes;
+                }
+            }
+            return tableExists;
+        }
+
+        //internal bool WaitforTable(CloudTableClient tableClient, string tableName, string waitChar, DateTime checkStart, int timeout)
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMHelper.cs]: remove if it is not used.

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMHelper.cs [hunk 11]
```diff
@@ -0,0 +1,722 @@
...
+                        var results = table.ExecuteQuery(query);
+                        if (results.Count() > 0)
+                        {
+                            tableExists = true;
+                            break;
+                        }
+                    }
+
+                    WriteHost(WaitChar, newLine: false);
+                    Thread.Sleep(5000);
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMHelper.cs]: use TestMockSupport.Delay(5000); instead

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMHelper.cs [hunk 12]
```diff
@@ -0,0 +1,722 @@
...
+                        var results = perfCounterTable.ExecuteQuery(new TableQuery() { FilterString = query });
+                        if (results.Count() > 0)
+                        {
+                            tableExists &= true;
+                            break;
+                        }
+                        else
+                        {
+                            WriteHost(waitChar, newLine: false);
+                            Thread.Sleep(5000);
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMHelper.cs]: use TestMockSupport.Delay(5000); instead

File: src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMExtensionConstants.cs [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+                    new PerformanceCounter() { counterSpecifier="\\NetworkInterface\\PacketsTransmitted",sampleRate = "PT15S"},
+                    new PerformanceCounter() { counterSpecifier="\\NetworkInterface\\PacketsReceived",sampleRate = "PT15S"},
+                    new PerformanceCounter() { counterSpecifier="\\NetworkInterface\\TotalRxErrors",sampleRate = "PT15S"},
+                    new PerformanceCounter() { counterSpecifier="\\NetworkInterface\\TotalTxErrors",sampleRate = "PT15S"},
+                    new PerformanceCounter() { counterSpecifier="\\NetworkInterface\\TotalCollisions",sampleRate = "PT15S"},
+                }
+            }
+        };
+
+        //internal static string GetStorageAccountFromUri(string uri)
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/AEM/AEMExtensionConstants.cs]: remove.

**Comment by hovsepm** [comment]
@MSSedusch  Please add license header to all files in the PR

**Pull request merged**